### PR TITLE
npc shop cash exploit warning

### PR DESCRIPTION
**Pull request by idontreallywolf**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

Send warning in console to warn about possible cash exploit.

**Issues addressed:** #3339 

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

**Review by omarcopires — APPROVED** [review]

**Comment by omarcopires** [comment]
This is a very useful check, but it would be interesting to check between all the npcs and also show the name of the NPC in Warning.

File: data/npc/lib/npcsystem/modules.lua [hunk 1]
```diff
@@ -809,6 +809,9 @@ if Modules == nil then
 			if shopItem == nil then
 				self.npcHandler.shopItems[#self.npcHandler.shopItems + 1] = {id = itemid, buy = cost, sell = -1, subType = itemSubType, name = realName or ItemType(itemid):getName()}
 			else
+				if cost < shopItem.sell then
+					print('[NpcSystem - ShopModule::addBuyableItem] Buy price lower than sell price ('.. shopItem.name ..')')
```
Comment (EPuncker) [inline data/npc/lib/npcsystem/modules.lua:813]:
```suggestion
					print("[Warning : " .. Npc():getName() .. "] NpcSystem: Buy price lower than sell price: (".. shopItem.name ..")")
```

File: data/npc/lib/npcsystem/modules.lua [hunk 2]
```diff
@@ -904,6 +907,9 @@ if Modules == nil then
 			if shopItem == nil then
 				self.npcHandler.shopItems[#self.npcHandler.shopItems + 1] = {id = itemid, buy = -1, sell = cost, subType = itemSubType, name = realName or ItemType(itemid):getName()}
 			else
+				if cost > shopItem.buy then
+					print('[NpcSystem - ShopModule::addSellableItem] Sell price higher than buy price ('.. shopItem.name ..')')
```
Comment (EPuncker) [inline data/npc/lib/npcsystem/modules.lua:911]:
```suggestion
					print("[Warning : " .. Npc():getName() .. "] NpcSystem: Sell price higher than buy price: (".. shopItem.name ..")")
```

**Review by EPuncker — CHANGES_REQUESTED** [review]

**Review by EPuncker — APPROVED** [review]

**Review by MillhioreBT — APPROVED** [review]
I do not see any reason to reject these changes, it will be of great help for the modification of our servers, respect to the Npc's ;)

**Review by EvilHero90 — APPROVED** [review]
good addition, don't see any reason as to why we shouldn't add this.

**Comment by Dip-Set1** [comment]
Found small bug, items with subtypes (mugs, vials, etc) and same item id, will create false report
```
[Warning : Sandra] NpcSystem:   Found duplicated item:   vial of oil,2006,20,11
[Warning : Sandra] NpcSystem:   Found duplicated item:   vial of slime,2006,12,4
[Warning : Sandra] NpcSystem:   Found duplicated item:   vial of urine,2006,10,13
[Warning : Sandra] NpcSystem:   Found duplicated item:   vial of water,2006,8,1
```

**Pull request merged**